sklearn/metrics/_ranking.py

```diff
@@ -372,15 +372,18 @@ def det_curve(
     >>> thresholds
     array([0.35, 0.4 , 0.8 ])
     """
+
+    xp, _, device = get_namespace_and_device(y_true, y_score)
     fps, tps, thresholds = _binary_clf_curve(
         y_true, y_score, pos_label=pos_label, sample_weight=sample_weight
     )
 
     # add a threshold at inf where the clf always predicts the negative class
     # i.e. tps = fps = 0
-    tps = np.concatenate(([0], tps))
-    fps = np.concatenate(([0], fps))
-    thresholds = np.concatenate(([np.inf], thresholds))
+    tps = xp.concat((xp.asarray([0.0], device=device), tps))
+    fps = xp.concat((xp.asarray([0.0], device=device), fps))
+    thresholds = xp.astype(thresholds, _max_precision_float_dtype(xp, device))
+    thresholds = xp.concat((xp.asarray([xp.inf], device=device), thresholds))
```

Comment:
Maybe we can avoid changing `thresholds` and instead adjust the `dtype` on the temp array that we are creating
```suggestion
    thresholds = xp.concat((xp.asarray([xp.inf], dtype=thresholds.dtype, device=device), thresholds))
```
Or do we need to adjust the `dtype` of `thresholds` specifically